src/modules/external_services/external_ip.py

```diff
@@ -0,0 +1,51 @@
+from collections import namedtuple
+
+import requests
+
+from .exceptions import ExternalProviderFailure, ExternalServiceFailure
+
+
+def generic_text_http_provider(url):
+    try:
+        return requests.get(url).text
+    except requests.ConnectionError as e:
+        raise ExternalProviderFailure(e)
+
+ExternalIPProvider = namedtuple('ExternalIPProvider', [
+    'provide', # provide the current address IP.
+    'priority' # it means: secure, no logs, etc.
+])
+
+CanHazIP = ExternalIPProvider(provide=generic_text_http_provider('https://canhazip.com'), priority=0)
+Ipify = ExternalIPProvider(provide=generic_text_http_provider('https://api.ipify.org'), priority=1)
```

Comment:
I don't think using priority fits here.
In my opinion, the service provider mechanism should be as easy as possible to use.
Maybe just use a list of them so first item is first priority?

Reply:
Sure, but if some user want to dynamically modify the providers and want to have his own to be used rather than the system ones, I'm not sure how it's going to be done.

Reply:
In that case we can build the failover list with his preference first and append the original list, filtered:
```python
failover_list = [preference] + [service for service in services if service != preference]
```